### PR TITLE
tiltfile: store normalized image name on dc_resource for matching against imagesByName

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @jazzdan,

Please review the following commits I made in branch maiamcc/tiltfile-err-messaging:

2f4f105ad55f4721499d3ccf612a2be2053ee1d2 (2019-01-25 15:20:52 -0500)
tiltfile: store normalized image name on dc_resource for matching against imagesByName

4c791725e180bd4e9140a0dc7627aacd89d517b0 (2019-01-25 11:37:38 -0500)
tiltfile: better err messaging for k8s validation

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/tiltfile/tiltfile_state.go [hunk 1]
```diff
@@ -184,11 +184,13 @@ func (s *tiltfileState) assembleK8s() (map[string]bool, error) {
 
 func (s *tiltfileState) validateK8s(r *k8sResource, assembledImages map[string]bool) error {
 	if len(r.entities) == 0 {
-		if len(r.providedImageRefs) == 0 {
-			return fmt.Errorf("resource %q: no matching resource", r.name)
-		}
+		if len(r.providedImageRefs) > 0 {
```
[the remark this answers is not in the document]
Comment (maiamcc) [reply internal/tiltfile/tiltfile_state.go]: ...wuh?

[the remark this answers is not in the document]
Comment (jazzdan) [reply internal/tiltfile/tiltfile_state.go]: Isn't this already in master as https://github.com/windmilleng/tilt/commit/b37584d6adbf30f8e11788a479d432b95e122567 ?

[the remark this answers is not in the document]
Comment (maiamcc) [reply internal/tiltfile/tiltfile_state.go]: oooh you're right

**Comment by maiamcc** [comment]
lol too many commits in this one: re-PR'd as #1066

**Pull request closed without merging**